packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -1518,14 +1522,11 @@ mixin WidgetInspectorService {
     if (creationLocation == null) {
       return false;
     }
-    return _isLocalCreationLocation(creationLocation);
+    return _isLocalCreationLocation(creationLocation.file);
   }
 
-  bool _isLocalCreationLocation(_Location? location) {
-    if (location == null) {
-      return false;
-    }
-    final String file = Uri.parse(location.file).path;
+  bool _isLocalCreationLocationImpl(String locationUri) {
+    final String file = Uri.parse(locationUri).path;
```

Comment:
it isn't the performance bottleneck but why do we perform the uri parsing before looking up in the _isLocalCreationCache? If we used the full locationUri, we would avoid the URI parsing for the common case where the URIs are already in the map.

Reply:
I don't believe we are.  Any code inside of `_isLocalCreationLocationImpl` will only get called if the cache doesn't have the value.  So that includes `Uri.parse`.  See the implementation of `isLocalCreationLocation`.